packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -281,59 +269,89 @@ class OnsetPicker extends React.Component {
-            { value: GranularityOptions.YEAR, display: intl.formatMessage({ id: 'Terra.onsetPicker.granularityYear' }) },
-            { value: GranularityOptions.MONTH, display: intl.formatMessage({ id: 'Terra.onsetPicker.granularityMonth' }) },
-            { value: GranularityOptions.DATE, display: intl.formatMessage({ id: 'Terra.onsetPicker.granularityDate' }) }]}
-          name={this.props.granularitySelectName}
+          className={cx('field-inline', 'granularity')}
           defaultValue={this.state.granularity}
           onChange={this.changeGranularity}
-          isInline
-        />
+          placeholder={intl.formatMessage({ id: 'Terra.onsetPicker.granularity' })}
+          isLabelHidden
+          label={intl.formatMessage({ id: 'Terra.onsetPicker.granularityLabel' })}
+          selectId={`${this.props.id}-granularity-select`}
+        >
+          <SelectField.Option
+            value={GranularityOptions.AGE}
+            display={intl.formatMessage({ id: 'Terra.onsetPicker.age' })}
+            key={GranularityOptions.AGE}
+          />
+          <SelectField.Option
+            value={GranularityOptions.YEAR}
+            display={intl.formatMessage({ id: 'Terra.onsetPicker.year' })}
+            key={GranularityOptions.YEAR}
+          />
+          <SelectField.Option
+            value={GranularityOptions.MONTH}
+            display={intl.formatMessage({ id: 'Terra.onsetPicker.month' })}
+            key={GranularityOptions.MONTH}
+          />
+          <SelectField.Option
+            value={GranularityOptions.DATE}
+            display={intl.formatMessage({ id: 'Terra.onsetPicker.date' })}
+            key={GranularityOptions.DATE}
+          />
+        </SelectField>
       );
     }
 
-    let ageSelect = null;
+    let ageInput = null;
     let ageUnitSelect = null;
     if (this.state.granularity === GranularityOptions.AGE) {
-      ageSelect = (
-        <NumberField
+      ageInput = (
+        <InputField
           className={cx('field-inline')}
-          data-terra-clinical-onset-picker="age"
-          min={0}
-          max={OnsetUtils.allowedAge(this.props.birthdate, this.state.ageUnit)}
-          step={1}
           defaultValue={this.state.age}
+          inputAttrs={{
+            type: 'number',
+            min: 0,
+            max: OnsetUtils.allowedAge(this.props.birthdate, this.state.ageUnit),
+            step: 1,
+          }}
+          inputId={`${this.props.id}-age-input`}
+          isLabelHidden
+          label={intl.formatMessage({ id: 'Terra.onsetPicker.ageLabel' })}
           onChange={this.changeAge}
-          isInline
         />
       );
 
       ageUnitSelect = (
         <SelectField
-          className={cx('field-inline')}
-          data-terra-clinical-onset-picker="age_unit"
-          options={OnsetUtils.allowedAgeUnits(this.props.birthdate, intl)}
-          defaultValue={this.state.ageUnit.toString()}
+          className={cx('field-inline', 'age-unit')}
```

Comment:
Can we create reusable logic here? `ageUnitSelect`, `monthSelect`, and `yearSelect` are almost one-one. 

We could create reusable function to render a `<SelectField />`. This function would accept prop values `defaultValue`, `value`, `onChange` and `selectId`. Lastly, a parameter `selectType` to represent "age", "month", or "year". Check this param to set classNames, translations, and the`<SelectField.Option />` children components. 

Thoughts?

Reply:
The only same thing is the SelectField and the isLabelHidden prop. Everything else would need to be different (albeit slightly) and thus included in the switch statement for the different select.  Doesn't seem like it would save many LOC or be that much clearer.